client/src/app/submission/submission.component.ts

```diff
@@ -274,11 +303,27 @@ export class SubmissionComponent implements OnInit, OnDestroy {
             const entitiesUrl = archiveSubmission['_links']['entities']['href'];
             this.archiveEntityDataSource = new IngestDataSource<ArchiveEntity>(this.ingestService, entitiesUrl, 'archiveEntities');
           }
-
         }
       );
   }
 
+  private initDataSources() {
+    const submissionTypes = ['biomaterials', 'processes', 'protocols', 'files', 'bundleManifests'];
+    submissionTypes.forEach(type => {
+      if (this[`${type}DataSource`]) {
+        // Ensure this is only called once.
+        // datasources should not be reinistatiated
+        // TODO: sort out other polling here and remove this check
+        return;
+      }
+      this[`${type}DataSource`] = new MetadataDataSource<any>(
+        (params) => this.ingestService.fetchSubmissionData(this.submissionEnvelopeId, type, params),
+        type
+      );
```

Comment:
very nice, i really like that you can pass the fetch function to the data source!